src/cli/test/parse-flags.spec.ts

```diff
@@ -30,556 +20,216 @@ describe('parseFlags', () => {
     expect(flags.unknownArgs[2]).toBe('test.spec.ts');
   });
 
-  it('should use cli args, no npm cmds', () => {
+  it('should parse cli for dev server', () => {
     // user command line args
     // $ npm run serve --port 4444
 
     // args.slice(2)
     // [ 'serve', '--address', '127.0.0.1', '--port', '4444' ]
 
-    args.push('serve', '--address', '127.0.0.1', '--port', '4444');
+    const args = ['serve', '--address', '127.0.0.1', '--port', '4444'];
 
-    const flags = parseFlags(args, sys);
+    const flags = parseFlags(args);
     expect(flags.task).toBe('serve');
     expect(flags.address).toBe('127.0.0.1');
     expect(flags.port).toBe(4444);
     expect(flags.knownArgs).toEqual(['--address', '127.0.0.1', '--port', '4444']);
   });
 
-  it('should use cli args first, then npm cmds', () => {
-    // user command line args
-    // $ npm run serve --port 4444
-
-    // npm script
-    // "serve": "stencil serve --address 127.0.0.1 --port 8888"
-
-    // args.slice(2)
-    // [ 'serve', '--address', '127.0.0.1', '--port', '8888', '4444' ]
-
-    // process.env.npm_config_argv
-    // {"remain":["4444"],"cooked":["run","serve","--port","4444"],"original":["run","serve","--port","4444"]}
-
-    args.push('serve', '--address', '127.0.0.1', '--port', '8888', '4444');
-
-    sys.getEnvironmentVar = (key: string): string => {
-      if (key === 'npm_config_argv') {
-        return JSON.stringify({
-          original: ['run', 'serve', '--port', '4444'],
-        });
-      }
-      return '';
-    };
-
-    const flags = parseFlags(args, sys);
-    expect(flags.task).toBe('serve');
-    expect(flags.address).toBe('127.0.0.1');
-    expect(flags.port).toBe(4444);
-  });
-
-  it('run stencil cmd from npm scripts', () => {
-    // user command line args
-    // $ npm run dev
-
-    // npm script
-    // "dev": "stencil build --dev --watch --serve"
-
-    // args.slice(2)
-    // [ 'build', '--dev', '--watch', '--serve' ]
-
-    // process.env.npm_config_argv
-    // {"remain":[],"cooked":["run","dev"],"original":["run","dev"]}
-
-    args.push('build', '--dev', '--watch', '--serve');
-
-    process.env = {
-      npm_config_argv: JSON.stringify({
-        original: ['run', 'dev'],
-      }),
-    };
-
-    const flags = parseFlags(args, sys);
-    expect(flags.task).toBe('build');
-    expect(flags.dev).toBe(true);
-    expect(flags.watch).toBe(true);
-    expect(flags.serve).toBe(true);
-  });
-
   it('should parse task', () => {
-    args[0] = 'build';
-    const flags = parseFlags(args, sys);
+    const flags = parseFlags(['build']);
     expect(flags.task).toBe('build');
   });
 
   it('should parse no task', () => {
-    args[0] = '--flag';
-    const flags = parseFlags(args, sys);
+    const flags = parseFlags(['--flag']);
     expect(flags.task).toBe(null);
   });
 
-  it('should parse build flag to true', () => {
```

Comment:
this test (and many others as well) is essentially completely redundant given our exhaustive testing of all boolean flags with `describe.each(BOOLEAN_CLI_ARGS`, below. While I was in here I wanted to remove them so that this file can be smaller, more focused, and easier to read. We have had a few bugs and edge cases crop up recently related to arg parsing, and since we're maintaining our own library for doing this I want to try to make sure this file is easy to read and understand, has exhaustive tests of all the supported arguments, and so on.